lib/controllers/habit_controller.dart

```diff
@@ -0,0 +1,325 @@
+import 'dart:ffi';
+import 'package:flutter/cupertino.dart';
+import 'package:intl/intl.dart';
+import 'package:spark/postgres_functions.dart';
+
+//habits: habit_id, user_id, title, note, start_date, end_date, frequency, reminders, reminder_message, target_type, category, quantity
+//activities: user_id, habit_id, timestamp, quanity
+
+//Homepage methods
```

Comment:
I don't see a class for "HabitController", are these all just floating methods? These should be wrapped in a class that implements "ChangeNotifier"